zephyr/lib/regions_mm.c

```diff
@@ -1,53 +1,428 @@
+					allocation->allocation_size);
+			} else {
+				retval = sys_mem_blocks_free(
+					heap->physical_blocks_allocators[i],
+					allocation->allocation_size, (void **)&ptr);
+			}
 			break;
-		case MEM_REG_ATTR_OPPORTUNISTIC_MEMORY:
-			vm_heaps[i].memory_caps = SOF_MEM_CAPS_RAM | SOF_MEM_CAPS_HP;
+		}
+	}
+
+	list_item_del(&allocation->node);
+	rfree(CONTAINER_OF(&allocation->node, struct active_allocation, node));
+
+	/* Scan for changes in physical pages mapping and
+	 * apply unmap if possible
+	 */
+	struct list_item *iterator_phys;
+	struct list_item *tmp;
+	struct list_item *iterator_activ;
+
+	list_for_item_safe(iterator_phys, tmp, &physical_mappings)
+	{
+		bool can_free = true;
+
+		list_for_item(iterator_activ, &heap->active_allocations)
+		{
+			struct active_physical_mapping *physical_maping = CONTAINER_OF(
+				iterator_phys, struct active_physical_mapping, node);
+			struct active_allocation *active_alloc = CONTAINER_OF(
+				iterator_activ, struct active_allocation, node);
+
+			if (is_range_in_range(
+				active_alloc->allocation_ptr,
+				active_alloc->allocation_size,
+				physical_maping->mapping_ptr,
+				physical_maping->mapping_size)) {
+
+				can_free = false;
+				break;
+			}
+		}
+		if (can_free) {
+			sys_mm_drv_unmap_region(CONTAINER_OF(iterator_phys,
+				struct active_physical_mapping, node)->mapping_ptr,
+				CONTAINER_OF(iterator_phys,
+					struct active_physical_mapping, node)->mapping_size);
+			list_item_del(iterator_phys);
+			rfree((void *)CONTAINER_OF(iterator_phys,
+					struct active_physical_mapping, node)->mapping_size);
 			break;
-		default:
-			return -EINVAL;
 		}
 	}
 
+	return retval;
+}
+
+/**
+ * @brief Reconfigure heap with given config
+ *
+ * This will destroy the heap from the pointer and recreate it
+ * using provided config. Individual region attribute is the
+ * "anchor" to the virtual space we use.
+ *
+ * @param heap Ptr to the heap that should be reconfigured
+ * @param cfg heap blocks configuration
+ * @param span allow contiguous on the reconfigured heap
+ *
+ * @retval heap_pointer Returns new heap pointer on success
+ * @retval NULL when reconfiguration failed
+ */
+struct virtual_memory_heap *reconfigure_heap(
+	struct virtual_memory_heap *heap, struct heap_config *cfg, bool span)
+{
+	struct virtual_memory_heap *retval = NULL;
+	uint32_t region_attribute = heap->virtual_region->attr;
+
+	if (vmh_free_heap(heap))
+		return retval;
+
+	return init_heap(cfg, region_attribute, span);
+}
+
+/**
+ * @brief Get default configuration for heap
+ *
+ * This will return config created based on region provided.
+ * It will try to split memory in even chunks and split it into
+ * bundles of size 64,128,...,1024
+ *
+ * @param region Virtual region which resources we will be using for heap
+ *
+ * @retval Returns default heap_config
+ */
+struct heap_config get_default_heap_config(const struct sys_mm_drv_region *region)
+{
+	int buffer_size = region->size/5;
```

Comment:
add a comment, why 5 and add spaces around `/`

Reply:
It is explained in function description, I expanded it a bit.